codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/SymbolVisitor.kt

```diff
@@ -211,7 +211,7 @@ open class SymbolVisitor(
     private fun handleOptionality(symbol: Symbol, member: MemberShape): Symbol =
         if (config.handleRequired && member.isRequired) {
             symbol
-        } else if (nullableIndex.isNullable(member)) {
+        } else if (nullableIndex.isMemberNullable(member, NullableIndex.CheckMode.CLIENT_ZERO_VALUE_V1_NO_INPUT)) {
```

Comment:
The server code generator is also using this code, so this may need to pass in a different `CheckMode` for its use-case?

Reply:
Yes, the server code generator will need to use`NullableIndex.CheckMode.SERVER` for this check.

Reply:
Let's add the `CheckMode` as a member of `SymbolVisitorConfig` so that it can be set correctly for the server codegen.

Reply:
I have pushed and update for this.